### PR TITLE
Add rename menu after the "Move method" refactoring

### PR DESCRIPTION
**Pull request by ilyailya**
### Motivation

The change adds a rename textfield after the refactoring have been applied. 
By the LSP spec, you can provide a command in the `CodeAction`, and it will be executed after all edits have been applied. 

```
export interface CodeAction {
	...
	
	/**
	 * A command this code action executes. If a code action
	 * provides an edit and a command, first the edit is
	 * executed and then the command.
	 */
	command?: Command;
}
```
I'm using this property of the code action to invoke the `sorbet.rename` command, which is also have been added in this change. The `sorbet.rename` commands calls the `editor.action.rename` command, which executes all needed requests to rename a symbol.

Unfortunately we can't call an `editor.action.rename` command directly, for more details please check here https://github.com/microsoft/vscode/issues/146767

https://user-images.githubusercontent.com/9540120/162059427-bf639cae-29c6-4dc4-930b-cc4894cc5a29.mov

The PR is best reviewed by commits

### Test plan

See included automated tests.

### CODE REVIEW

File: main/lsp/requests/code_action.cc [hunk 1]
```diff
@@ -142,15 +142,27 @@ unique_ptr<ResponseMessage> CodeActionTask::runRequest(LSPTypecheckerInterface &
                     }
                     auto action = make_unique<CodeAction>("Move method to a new module");
                     action->kind = CodeActionKind::RefactorExtract;
+
                     bool canResolveLazily = config.getClientConfig().clientCodeActionResolveEditSupport &&
                                             config.getClientConfig().clientCodeActionDataSupport;
+                    auto newModuleLoc = getNewModuleLocation(gs, *def, typechecker);
+                    auto renameCommand = make_unique<Command>("Rename Symbol", "sorbet.rename");
```
Comment (jez) [inline main/lsp/requests/code_action.cc:149]: What happens if the client doesn't support this command (e.g., old VS Code extension version, or using Vim or some other language client)

**Review by jez — APPROVED** [review]
One question about backwards compatibility / compatibility with other language clients.

File: main/lsp/requests/code_action.cc (hunk 1 above)
Comment (ilyailya) [reply main/lsp/requests/code_action.cc:149]: It depends on the client, VSCode will show a warning popup like this, but the edits will be applied.
Screen Shot 2022-04-07 at 14 53 48

NeoVim's LSP client just ignores the command

https://user-images.githubusercontent.com/9540120/162217207-b31c4625-829c-4750-a278-4c8f58f23274.mov

Comment (jez) [reply main/lsp/requests/code_action.cc:149]: Perfect thanks

**Pull request merged**